core/src/worker/workflow/machines/timer_state_machine.rs

```diff
@@ -73,13 +74,17 @@ pub(super) fn new_timer(attribs: StartTimer) -> NewMachineWithCommand {
 impl TimerMachine {
     /// Create a new timer and immediately schedule it
     fn new_scheduled(attribs: StartTimer) -> (Self, Command) {
+        let user_metadata = attribs.summary.clone().map(|x| UserMetadata {
+            summary: Some(x),
+            details: None,
+        });
         let mut s = Self::new(attribs);
         OnEventWrapper::on_event_mut(&mut s, TimerMachineEvents::Schedule)
             .expect("Scheduling timers doesn't fail");
         let cmd = Command {
             command_type: CommandType::StartTimer as i32,
-            attributes: Some(s.shared_state().attrs.into()),
-            user_metadata: Default::default(),
+            attributes: Some(s.shared_state().attrs.clone().into()),
```

Comment:
Without the clone, I get the error 
```
cannot move out of a shared reference
move occurs because value has type `temporal_sdk_core_protos::coresdk::workflow_commands::StartTimer`, which does not implement the `Copy` traitrustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic message [0]?0#file:///Users/andrewyuan/temporal/core/core/src/worker/workflow/machines/timer_state_machine.rs)
```

Looks like we clone the attrs of Activity https://github.com/yuandrew/sdk-core/blob/dd37e2946a385090ded0fc3312180027002d4b16/core/src/worker/workflow/machines/activity_state_machine.rs#L134

Reply:
hmm, curious why we're even getting this error in the first place.. the line I added for `user_metadata` doesn't involve `s` directly, but maybe because I'm cloning `attribs.summary`, it messes with the borrow checker, when s is created?`let mut s = Self::new(attribs);`

Is there a different workaround here, or would we have to clone (or implement `Copy`) here?

Reply:
Ah, yeah I didn't see it getting moved later. This clone is fine to keep then, but we could change the line above to also `take` the summary. 

And yeah, in general I want to reduce how many places lots of various strings get cloned, like the place you linked, but that's a bigger project. We may not ever do it since ultimately it's not that big of a deal but, I want to avoid adding more spots where we unnecessarily clone.